core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -945,7 +960,8 @@ private void evaluateTriggerable(Triggerable t, TreeReference anchorRef) {
      * when we support dependent XPath Steps (IE: /path/to//)
      */
     public void addChildrenOfReference(TreeReference original, Vector<TreeReference> toAdd) {
-        for (TreeReference ref : exprEvalContext.expandReference(original)) {
+        Vector<TreeReference> refs = exprEvalContext.expandReference(original);
```

Comment:
Rather, what assumptions about structure homogeneity can we assume at this point? If we can assume that the target of triggers will be homogeneous, we can just cache the template and be done with this child traversal funny business (notice that we store generic refs to all the nodes collected this way, so we end up with duplicate references!)